### ISSUE TITLE
App not redirecting after accepting permissions

### ISSUE THREAD

**Issue by paulcanning** [ohmybrew/laravel-shopify#78]
When installing an app, and accepting the permissions, the user is redirected to the app domain, and I believe should then be redirected again to shopify, but that is not happening.

The JS code in the page is as follows:

```
                ShopifyApp.init({
                    apiKey: '<MYKEY>',
                    shopOrigin: 'https://my-shop.myshopify.com',
                    debug: false,
                    forceRedirect: true
                });      
```

If I hit refresh, the page loads and then does the redirect.

**Comment by ohmybrew** [comment]
Is there an error in Shopify's console.

I see the app load fullscreen on my end, then it goes into the iframe. Do you know if that bit of code is running for sure?

**Comment by paulcanning** [comment]
Try adding your app by going to the app's URL

**Comment by ohmybrew** [comment]
@paulcanning Sorry man, still a tad confused ha. So steps is:

1. Go to apps main url directly
2. It shows app page, but does not redirect to Shopify iframe?

**Comment by paulcanning** [comment]
Thats correct.

Still seeing this issue.

I'll try see if i can step through it, but yea, add the app by visiting the app directly, it'll go to Shopify to authorise, and send you back to the app home page, where it stays.

Refreshing the home page then does the redirect, and any subsequent visit to the app home page does the redirect.

**Comment by ohmybrew** [comment]
Interesting will try that out and see what I can find out.

**Comment by ohmybrew** [comment]
@paulcanning Can you let me know if this is the steps? https://streamable.com/b21x7

If not let me know! Trying to reproduce but maybe I'm not understanding.

**Comment by paulcanning** [comment]
Yea that's exactly it. But for some reason the redirect back to shopify doesn't work

**Comment by ohmybrew** [comment]
When it *doesn't* redirect for you, can you view source and see if the esdk
is loading in the layout? I wonder if it's some sort of odd bug there.

**Comment by ohmybrew** [comment]
@paulcanning to add to above, any errors in the console of the browser?

**Comment by paulcanning** [comment]
Finally got round to testing this again, running through a fresh install.

Whilst I am using my own packages (using your Shopify API package) - It uses the same approach for redirecting etc

Here is what I see in the console:

```
ShopifyApp detected that it was not loaded in an iframe and is redirecting to: https://<my-shop>.myshopify.com/admin/apps/340d5f20ef3d9cab4a233250e59d077d/
app.js?2018083008:1 ShopifyApp client sent {"message":"Shopify.API.initialize","data":{"debug":true,"apiKey":"<API_KEY>"}} to https://<my-shop>.myshopify.com
2app.js?2018083008:1 ShopifyApp client received [object Object] from unknown origin http://<my-app>.test. Expected https://<my-shop>.myshopify.com.
```

**Comment by ohmybrew** [comment]
> ShopifyApp client received [object Object] from unknown origin http://<my-app>.test. Expected https://<my-shop>.myshopify.com.

That last bit is odd? Not sure I understand it. Does the view-source show the ESDK loading? Line 2 of the console seems to suggest so..

**Comment by ohmybrew** [comment]
@paulcanning Still an issue, cant reproduce at all :( any note on above comment?

**Comment by hakimihamdan88** [comment]
i'm having this weird bug too. white blank page. need to hit refresh, then will redirect.

**Comment by paulcanning** [comment]
@ohmybrew yup, it is still happening sporadically, but I'm currently off that project for the time being.

**Comment by ohmybrew** [comment]
I've moved fullpage_redirect view file to use Shopify's own JS for redirection. Seems to be working fine for me. Not sure if anyone can pull the `issue-78` branch (v4) and see if the issue is solved. Else, simply take the code from [here to test](https://github.com/ohmybrew/laravel-shopify/blob/3790b1856b7a3fbe6b54e6c2669ed4a0af3b06d9/src/ShopifyApp/resources/views/auth/fullpage_redirect.blade.php) by adding `fullpage_redirect.blade.php` to `resources/views/vendor/shopify-app/auth/`.
